src/Universum/List/Safe.hs

```diff
@@ -25,8 +28,20 @@ import Universum.Monad (Maybe (..), Monad (..))
 -- >>> import Universum.Bool (Bool (..), not)
 -- >>> import Universum.Foldable (length)
 -- >>> import Universum.Function (($))
+-- >>> import Universum.List.Reexport (head)
 -- >>> import Universum.Print (print)
 
+{- | For safe work with lists using functinons for 'NonEmpty'.
+
+>>> viaNonEmpty head [1]
+Just 1
+>>> viaNonEmpty head []
+Nothing
+
+-}
+viaNonEmpty :: (NonEmpty a -> a) -> [a] -> Maybe a
+viaNonEmpty f = fmap f . nonEmpty
+
```

Comment:
I think it's better to add `{-# INLINE viaNonEmpty #-}` to this function to make it more efficient.